### ISSUE TITLE
Unhelpful error messages (bad developer experience)

### ISSUE THREAD

**Issue by KillyMXI** [AvaloniaUI/Avalonia#9256]
**Describe the bug**

During the app development there were few occasions when Axaml editor reports "Invalid Markup" errors, but they are actually not about markup and absolutely unhelpful to identify the actual cause, resulting in frustrating debugging sessions.

**To Reproduce**

1. Work on a project (about 20 views) long enough, it will happen eventually.

I can't give more condensed steps exactly because I don't know yet where the issue is. That's exactly what I'm complaining about.

**Expected behavior**

Error message or Avalonia Diagnostics entry that will be more helpful to pinpoint what exactly is missing.
Better way of reporting errors in code-behind is needed.

All I got is that the error happened when evaluating `MainWindow.axaml`.
If it can't be linked to a certain line in the axaml file, the error message text should include more information.
The evaluator code should customize error messages it rethrows to be more specific.

**Screenshots**

![image](https://user-images.githubusercontent.com/13851064/197188631-512cc621-0d0c-4793-9471-0c3d51f5f6f6.png)

![image](https://user-images.githubusercontent.com/13851064/197188690-bc6b837f-8d0b-4b1f-8300-2df814135690.png)

![image](https://user-images.githubusercontent.com/13851064/197188955-8cc78ba1-3c7c-44e3-b0c1-060a65609515.png)

**Desktop (please complete the following information):**

- OS: Windows 10
- Microsoft Visual Studio Community 2022 Version 17.3.6
- Version 0.10.18

**Comment by timunie** [comment]
Enable compiled Bindings if you have not already. Often these help to debug. 

https://docs.avaloniaui.net/docs/data-binding/compiledbindings

**Comment by kekekeks** [comment]
I don't think we have `System.Xaml.XamlException` class used anywhere in our codebase

**Comment by kekekeks** [comment]
https://github.com/AvaloniaUI/AvaloniaVS/blob/f18343c6fbe8fed90b8d3743b8c21483818f2eeb/AvaloniaVS.Shared/Services/PreviewerProcess.cs#L444 - we should probably somehow expose the stack trace here as well.

The problem is that exception happens when XAML is already parsed and compiled into MSIL. 
So there is no reliable way to extract the line info at that point.

**Comment by kekekeks** [comment]
We might need to add debug information support for System.Reflection.Emit backend as well - https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/a263ywy6(v=vs.100)?redirectedfrom=MSDN
